src/nvim/os/signal.c

```diff
@@ -45,6 +46,14 @@ void signal_init(void)
   signal_watcher_init(&main_loop, &spwr, NULL);
   signal_watcher_start(&spwr, on_signal, SIGPWR);
 #endif
+
+#ifdef SIGCHLD
```

Comment:
Not that it matters too much in this, but to indicate that this should be more like a "clean slate", it might be better to move this code to the top of the function.